### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by AmaliaGM**
## Project requirements

- Add Redux Toolkit(npm install react-redux @reduxjs/toolkit).

- Structure your application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files. Your files/folder structure could look like this:

...
/src
    |---/components
    |---/redux
        |--- /books
                | books.js
        |--- /categories
                | categories.js
        | configureStore.js
    | index.js
- Write your book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

- Define action types for adding and removing a book.
- Set the initial state to be an empty array of books.
- Export Action Creators for your actions.
- Write your reducer and export it as default.
- Define state changes for the actions that you created.
- In case of unknown action - return the current state.
- Write your categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

- Define an action type for checking the status.
- Set the initial state to be an empty array of categories.

- Write your reducer and export it as default.
- For the check status action return a string "Under construction".
- In case of unknown action - return the current state.
- Configure the Redux Store (/src/redux/configureStore.js):

- Importing the necessary methods from Redux Toolkit.
- Combine both reducers into a root reducer by using configureStore function.

### CODE REVIEW

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @AmaliaGM 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check the comments under the review.

```.js
//  BookStore/src/Redux/Book/reducer.js 


const ADD_BOOKS = 'bookstore/books/ADD_BOOKS';

export const addBook = (payload) => ({
  type: ADD_BOOKS,
  payload,
});

const initialState = [];
const bookReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_BOOKS:
      return [...state, action.payload];
    default:
      return state;
  }
};

export default bookReducer;
```
- [ ] Kindly define an action type and an action creator for removing a book. I can see only the action type and creator for adding a book. also, add the logic for removing the book in your reducer.

- [ ] Kindly make sure you configure your store using `@reduxjs/toolkit.` as it is mentioned in the requirements

```js
// BookStore/src/Redux/ConfigureStore.js 


import { createStore, combineReducers } from 'redux';
// import logger from 'redux-logger';
import bookReducer from './Book/reducer';
import categoryReducer from './Categories/categories';

const reducers = combineReducers({
  bookReducer,
  categoryReducer,
});

const store = createStore(reducers);

export default store;
```


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Redux/ConfigureStore.js [hunk 1]
```diff
@@ -8,6 +8,6 @@ const reducers = combineReducers({
   categoryReducer,
 });
 
-const store = createStore(reducers);
+const store = configureStore(reducers);
```
Comment (Ismailco) [inline src/Redux/ConfigureStore.js]: - [ ] good job working on the previous reviewer suggestions however to use configure store you can remove the combineReducers and do something like this 
```
const store = configureStore({
  reducer: {
    bookReducer: bookReducer,
    categoryReducer: categoryReducer 
  }
});
```

File: src/Redux/Book/reducer.js [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
+export const filterBooks = (payload) => ({
+  type: FILTER_BOOKS,
+  payload,
+});
 const initialState = [];
+
 const bookReducer = (state = initialState, action) => {
   switch (action.type) {
+    case FILTER_BOOKS:
+      return { ...state, filterBooks: action.filter };
```
Comment (Ismailco) [inline src/Redux/Book/reducer.js]: - [ ] Filter_books is actually removing a book and it should be using the filter method, kindly change it so it works as expected. your code may be like `return state.filter(book => book.id !== id)`

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @AmaliaGM, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights

✔️  No linter errors
✔️ Used Github Flow
✔️ Work documented professionally

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Redux/Book/reducer.js [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+export const deleteBook = (payload) => ({
+  type: DELETE_BOOK,
+  payload,
+});
 const initialState = [];
+
 const bookReducer = (state = initialState, action) => {
   switch (action.type) {
+    case DELETE_BOOK:
+      return { ...state, deleteBook: action.filter };
```
Comment (Ismailco) [inline src/Redux/Book/reducer.js]: - [ ] DELETE_BOOK should be using the filter method. for the filter to work it should have a function/parameter that filter the array, to do so you can add `(book => book.id !== id)` as the body of the filter method, also you need to return only the one filtred book not the whole state so you can remove the specific book. kindly change your code so it works properly

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @AmaliaGM, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Redux/Book/reducer.js [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
 
 export const addBook = (payload) => ({
   type: ADD_BOOKS,
   payload,
 });
-
+export const deleteBook = (payload) => ({
+  type: DELETE_BOOK,
+  payload,
+});
```
Comment (ahmetbozaci) [inline src/Redux/Book/reducer.js:11]: ### OPTIONAL
_I make it optional but this is a requirement that is not mentioned in the last reviewer. Please fix it._
- [ ] I can't delete a book by clicking on the 'remove' button on the page. Kindly fix it.

**Review by ahmetbozaci — APPROVED** [review]
# 🥳 🎉🥇 💯  💥 🎆 🎈 STATUS: APPROVED 🥳 🎉  🥇  💯  💥   🎆 🎈 🎉 



@AmaliaGM  ✋🏼 
#### Your project is great. 🎉 I approve of it. There is nothing else to say other than... it's time to merge it. You can submit this project on your dashboard.
___ 

### 💯 Highlights 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️

___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
@AmaliaGM @AmaliaGM 
- [ ] Kindly use descriptive names for branch names. [Requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md#:~:text=Respect%20the%20rules%20of%20the%20code%20review%20flow%20that%20your%20team%20follows.%20Use%20descriptive%20names%20for%20your%20branches%20(e.g.%2C%20refactor%2Dauthentication%2C%20user%2Dcontent%2Dcache%2Dkey%2C%20make%2Dretina%2Davatars)%20and%20make%20sure%20that%20your%20%27feature%20branch%27%20is%20up%2Dto%2Ddate%20with%20the%20%27main%20branch%27%20in%20your%20repository.)
- [ ] ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ 

- Your commit messages should always start with capital letters and they should be an imperative tense. I see some of them are not like that. Instead of writing `Added style or Created the file` kindly write `Add style or Create the file`
- ***I write it as optional. But this is a [general requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md#:~:text=The%20commit%20message%20should%20be%20written%20in%20the%20imperative%20tense%20and%20capitalized) for every project. So you should pay attention to this in your next projects.***
- Kindly change them by using `git-rebase`. If you want to know how to change ı leave a few resources.
- [Video - How to change commit messages](https://www.loom.com/share/00771f369b7b47fc968c019ecf0d3ba6)
- [Article - How to create good commit messages](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8/)

⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ 
___

### Cheers and Happy coding! 💻 👏
aproved


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Comment by AmaliaGM** [comment]
@ahmetbozaci Thanks so much! 🙏🥵 😁

**Pull request merged**